core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -286,8 +287,9 @@ object SparkSubmit {
 
     // The following modes are not supported or applicable
     (clusterManager, deployMode) match {
-      case (MESOS, CLUSTER) =>
-        printErrorAndExit("Cluster deploy mode is currently not supported for Mesos clusters.")
+      case (MESOS, CLUSTER) if args.isPython =>
+        printErrorAndExit("Cluster deploy mode is currently not supported for python " +
+          "on Mesos clusters.")
```

Comment:
for python applications on Mesos clusters
(to be consistent with the warning message for standalone cluster python)